zanata-war/etc/scripts/Mysql2Postgresql.groovy

```diff
@@ -0,0 +1,457 @@
+// indicated
+// Returns an object indicating if the script should proceed
+def initialize(args) {
+    def cli = new CliBuilder(usage: 'groovy Mysql2Postgresql.groovy [options]', header: 'Options:')
+
+    cli.mysqlhost(args:1, argName: 'mysqlhost', 'Host name or IP address of the mysql database (default: localhost)')
+    cli.mysqlport(args:1, argName: 'mysqlport', 'Open port to connect to the mysql database (default: 3306)')
+    cli.mysqlschema(args:1, argName: 'mysqlschema', 'MySql schema to migrate')
+    cli.mysqluser(args:1, argName: 'mysqluser', 'User for the mysql database (default: "")')
+    cli.mysqlpassword(args:1, argName: 'mysqlpassword', 'Password for the mysql database (default: "")')
+
+    cli.psqlhost(args:1, argName: 'psqlhost', 'Host name or IP address of the postgresql database (default: localhost)')
+    cli.psqlport(args:1, argName: 'psqlport', 'Open port to connect to the postgresql database (default: 5432)')
+    cli.psqldb(args:1, argName: 'psqldb', 'Postgresql database to migrate data into')
+    cli.psqluser(args:1, argName: 'psqluser', 'User for the portgresql database (default: "")')
+    cli.psqlpassword(args:1, argName: 'psqlpassword', 'Password for the postgresql database (default: "")')
+
+    cli.script(args:1, argName: 'script', 'Script file with the PostgreSQL DDL')
+    cli.h(longOpt: 'help', 'Print this message')
+
+    def options = cli.parse(args)
+
+    def cliArgs = options.arguments()
+    if(options.help || args.length <= 0) {
+        cli.usage()
+        return false
+    }
+
+    // If the option to generate a script are given
+    if(options.script) {
+        generateDDL = true
+        ddlFile = new File(options.script).newWriter()
+    }
+
+    def mysqlhost = options.mysqlhost ?: "localhost"
+    def mysqlport = options.mysqlport ?: "3306"
+    def mysqlschema = options.mysqlschema
+    def mysqluser = options.mysqluser ?: ""
+    def mysqlpassword = options.mysqlpassword ?: ""
+
+    def psqlhost = options.psqlhost ?: "localhost"
+    def psqlport = options.psqlport ?: "5432"
+    def psqldb = options.psqldb
+    def psqluser = options.psqluser ?: ""
+    def psqlpassword = options.psqlpassword ?: ""
+
+    if(!mysqlschema) {
+        println "You must provide a mysql schema to migrate from"
+        cli.usage()
+        return false
+    }
+    if(!psqldb && !generateDDL) {
+        println "You must provide a PostgreSql schema to migrate into"
+        cli.usage()
+        return false
+    }
+
+    mysqldb = Sql.newInstance("jdbc:mysql://${mysqlhost}:${mysqlport}/${mysqlschema}",
+        "${mysqluser}", "${mysqlpassword}", "com.mysql.jdbc.Driver")
+    mysqlMetadata = mysqldb.connection.metaData
+    // tell mysql driver to conserve memory (streaming resultset)
+    mysqldb.resultSetConcurrency = ResultSet.CONCUR_READ_ONLY
+    mysqldb.withStatement{stmt -> stmt.fetchSize = Integer.MIN_VALUE }
+
+    if(generateDDL) {
```

Comment:
Just wanted to keep it consistent as in all other places I wrote an 'if(generateDDL) {} else {}' structure. Makes it more readable imo